src/DataTool.php

```diff
@@ -181,7 +168,35 @@ public function generateData()
             }
         }
     }
-    
+
+    /**
+     * Generates data for provided field name based on field definition in vardef.php for proper module
+     *
+     * @param $field string - field name
+     */
+    protected function generateFieldData($field)
+    {
+        $type = (!empty($this->fields[$field]['dbType']))
+            ? $this->fields[$field]['dbType']
+            : $this->fields[$field]['type'];
+        $GLOBALS['fieldData'] = $this->fields[$field];
+
+        /* There are 3 unique parts to the seed: the Module name,
+         * the count of the record, and the name of the field.
+         * Using these 3 things should keep our seed unique enough.
+         */
+        $seed = $this->generateSeed($this->module, $field, $this->count);
+        $value = $this->getData($field, $type, $this->fields[$field]['type'], $seed);
```

Comment:
Let's delete `getData` method and place it's body here.

Regarding the size of the method: you can rewrite getData in the following form:
```php
foreach ([$this->module, 'default'] ...
    foreach ([$field, $type, $this->fields[$field]['type']]...
      if (isset...
```